tests/pytests/test_http_monitor.py

```diff
@@ -0,0 +1,104 @@
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+# =============================================================================
+
+import sys
+sys.path.insert(0, "")  # for running from cmake
+import pytest
+import zlib
+import json
+from datetime import datetime
+try:
+    import urllib.request as urllib_req
+except ImportError:
+    import urllib2 as urllib_req
+from statistics_checker import MonitorStatsChecker
+from statistics_checker import categories_combination
+
+import elliptics
+
+path_categories = {"/cache": elliptics.core.monitor_stat_categories.cache,
+                   "/io": elliptics.core.monitor_stat_categories.io,
+                   "/commands": elliptics.core.monitor_stat_categories.commands,
+                   "/backend": elliptics.core.monitor_stat_categories.backend,
+                   "/stats": elliptics.core.monitor_stat_categories.stats,
+                   "/procfs": elliptics.core.monitor_stat_categories.procfs,
+                   "/top": elliptics.core.monitor_stat_categories.top,
+                   "/all": elliptics.core.monitor_stat_categories.all}
+
+
+@pytest.fixture(scope="session")
+def address(servers):
+    address = {}
+    address['remote'], address['port'] = zip(servers.remotes, servers.monitors)[0]
+    address['remote'] = address['remote'].split(':')[0]
+    return address
+
+
+class TestHttpMonitor:
+
+    @pytest.mark.parametrize("path", path_categories.keys())
+    @pytest.mark.usefixtures("servers")
+    def test_http_statistics_by_path(self, address, routes, backends_combination, path):
+        '''Requests all possible combination of path and backends one by one and checks statistics'''
+
+        start = datetime.now()
+        json_stat = self.__get_statistics_by_path(address=address,
+                                                  path=path,
+                                                  backends=backends_combination)
+        end = datetime.now()
+
+        checker = MonitorStatsChecker(json_statistics=json_stat,
+                                      time_period=(start, end),
+                                      categories=path_categories[path],
+                                      routes=routes,
+                                      backends_combination=backends_combination)
+        checker.check_json_stat()
+
+    @pytest.mark.parametrize("categories", categories_combination())
+    @pytest.mark.usefixtures("servers")
+    def test_http_statistics_by_category(self, address, routes, backends_combination, categories):
```

Comment:
Let's also test http server with incorrect urls: non-integers in backends, backends without value (`?backends`/`&backends`), unknown/unused parameters,  non-integer values of `categories`, `categories` without value, a parameter with empty name.